src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java

```diff
@@ -169,4 +166,159 @@ public static String getCollection(Firestore database, String collectionToGetFro
        throw new FirebaseException(ServletConstantValues.UNABLE_TO_READ_FROM_FIRESTORE, e);
     }
   }
-} 
\ No newline at end of file
+
+  /**
+   * Adds support for infinite scroll by batching queries into pages of a limited size.
+   * Either we send in the first requested batch, or we send the next batch after one has 
+   * already been sent.
+   * @param database - The database to retrieve from.
+   * @param collectionToGetFrom - The name of the collection to retrieve the batch from.
+   * @param batchSizeLimit - The maximum size of the batch, which is the size of a new page of 
+   *   data in infinite scroll.
+   * @param lastDocID - The ID of the last object I sent from the most recent batch,
+   *   if null, then execute getFirstBatch().
+   * @param parameterToSortBy - The parameter I want to presort my collection by, 
+   *   if null then don't sort.
+   * @return - The json string representing the new batch of documents to be sent to the frontend.
+   * @throws FirebaseException
+   */
+  public static String getCollectionBatch(Firestore database, String collectionToGetFrom, 
+      int batchSizeLimit,  String lastDocID, String parameterToSortBy) 
+      throws FirebaseException {
+    // A batch size limit needs to be specified in order to make any query.
+    if (batchSizeLimit == 0) {
+      throw new FirebaseException(BATCH_SIZE_NOT_SPECIFIED);
+    }
+
+    CollectionReference collectionReference = database.collection(collectionToGetFrom);
+
+    // If the lastDocID is null, then we know this is the first batch to send.
+    // Else simply get the next batch in the collection.
+    if (lastDocID == null) {
```

Comment:
Use Optional to protect against NPEs.
https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html